wsgi: handle remote connection resets

stefannica opened the pull request:
While processing a WSGI request, if the remote client forcefully
resets the connection while the request data is being read from the
socket (e.g. HAProxy health check), the ECONNRESET socket error is
not properly handled and may end up being logged in strerr or in
a log file.

https://github.com/eventlet/eventlet/issues/446

temoto commented:
Do not call get_errno twice, performance police is sad.

Why do you think this condition doesn't deserve a line in log?

stefannica commented:
Thanks @temoto, I updated the PR.

> Do not call get_errno twice, performance police is sad.

Good point, thanks.

> 
> Why do you think this condition doesn't deserve a line in log?

This condition is ignored in the non-SSL case (see for instance [here](https://github.com/eventlet/eventlet/blob/master/eventlet/green/ssl.py#L247)), which is why this is only seen when SSL is enabled. Nevertheless, I added a log line for it, but only if debug is enabled. The point of this change is to keep the logs from being flooded.

temoto commented on eventlet/wsgi.py:
```diff
@@ -381,7 +381,12 @@ def _read_request_line(self):
             pass
         except socket.error as e:
-            if support.get_errno(e) not in BAD_SOCK:
+            erno = support.get_errno(e)
```
Why does this line introduce error on line 389?

stefannica replied:
Yeah, that would be my typo. Thank you for spotting that.

temoto replied:
Nah, typo is not the point. There will be more typos, it's fine. Problem is that tiny typo coerces to another name.

This line introduces error because name `erno` is only 1 letter distance away from `errno`. Defensive coding says make things sufficiently distinct, to avoid errors in writing and, perhaps more importantly, in reading.

If it was called `bQ8kk` or `current_errno` or something else distinctive, then it would be hard to typo into existing name even in the middle of hangover night.

Now latest patch shadows `errno` module name. It works, but name shadowing is road to trouble in general.

If you are sick from picky review, please let me know, I will change code and keep your authorship to commit history.

Thanks for your work.

stefannica replied:
I completely agree. It's just so easy to lose track of these common sense rules when you're going for a quick patch, so thank you for taking the time to offer your insight and for reminding me.

stefannica commented:
@temoto, could you give this another look ? Thanks.
